### ISSUE TITLE
The MongoClient is never closed, is this optional?

### ISSUE THREAD

**Issue by frouo** [vercel/mongodb-starter#24]
Hi there,

I'm working on implementing MongoDB into my Next.js project. Thanks for this starter.

I was curious why is the Mongo client never closed anywhere? is `client.close()` optional even for production?

As far as I understand, there is no need to close in `development` mode since the module is kept across reloads. But what about production especially when the project is hosted on Vercel (which uses serverless functions)?

Thanks in advance for your help 🙏

```
import { MongoClient } from 'mongodb';

export default async function handler(req, res) {
  const client = await MongoClient.connect(process.env.MONGODB_URI);
  const db = client.db();

  const collection = db.collection('users');
  const users = await collection.find().toArray();

  await client.close() // <- is this important for production?

  res.status(200).json(users);
}
```

**Comment by smaeda-ks** [comment]
@frouo have you tried setting the `maxIdleTimeMS` option?
https://www.mongodb.com/docs/drivers/node/current/fundamentals/connection/connection-options/

> Specifies the amount of time, in milliseconds, a connection can be idle before it's closed. Specifying 0 means no minimum.

and its default is `0`, as per their documentation above. So maybe setting it to a reasonable value (and adjust `maxPoolSize`)?

**Comment by forbesd7** [comment]
@frouo  
I was running into this issue as well - I don't think you need to close the connections as it _seems_ like they close automatically after not being used for a certain amount of time. Will continue to monitor connections as i test and see if this code should be adjusted to close the connection with each request.

I changed the mongodb.ts file to the following code and this seems to have fixed my issue. 


([Taken from this site](https://hevodata.com/learn/next-js-mongodb-connection/))

```
import { MongoClient } from 'mongodb';

declare global {
  var _mongoClientPromise: Promise<MongoClient>;
}


const uri = process.env.MONGODB_URI || '';
const options = {};

let client;
let clientPromise;

if (!process.env.MONGODB_URI) {
  throw new Error('Add Mongo URI to .env.local');
}

if (process.env.NODE_ENV === 'development') {
  if (!global._mongoClientPromise) {
    console.log('getting client');
    client = new MongoClient(uri, options);
    global._mongoClientPromise = client.connect();
  }
  clientPromise = global._mongoClientPromise;
} else {
  client = new MongoClient(uri, options);
  clientPromise = client.connect();
}

export default clientPromise;
```


Just a note - this is ONLY when i'm testing on local. No idea how so many connections were being made. You can see the giant reduction in connections after I change to this code.

<img width="341" alt="Screenshot 2023-06-29 at 13 01 22" src="https://github.com/vercel/mongodb-starter/assets/54018183/4b4bb0ca-9af7-461b-b82d-12b29619ca0d">

**Comment by smaeda-ks** [comment]
we also have this example, fyi: https://github.com/vercel/mongodb-starter/blob/main/lib/mongodb.ts

**Comment by forbesd7** [comment]
@smaeda-ks Right, i was using that but it was creating a bunch of connections - i removed that code and added the one above which reduced the connection amount by a lot.

**Comment by smaeda-ks** [comment]
Ah, ok. I'll have some tests later and see if I can fix that.

**Comment by frouo** [comment]
@smaeda-ks no, I haven't touched to `maxIdleTimeMS` or any other setting as I would expect mongodb to work kinda out-of-the-box.

I was wondering, does the [Vercel <> MongoDB integration](https://vercel.com/integrations/mongodbatlas) fix this connection issue somehow? Does it auto-config mongodb atlas to "fit" the nextjs serverless environment?

**Comment by frouo** [comment]
Sorry, I'm rephrasing. I don't expect MongoDB to work out-of-the-box with Next.js, but I'm not confident in changing MongoDB configurations when no tutorial does (especially considering the Vercel MongoDB starter).

This actually surprises me because, as I mentioned earlier, as soon as I released the sample code above, I received a "connections exceeded" warning from MongoDB 🤷‍♂️ It's clear that using MongoDB out-of-the-box does not work, but there is no documentation on Vercel on how to configure MongoDB correctly.

Is `maxIdleTimeMS` the way to go? If so, what value does Vercel recommend? Thank you very much :)
